seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/Firefox.scala

```diff
@@ -0,0 +1,55 @@
+package org.scalajs.jsenv.selenium
+
+import org.scalajs.core.tools.io._
+
+import org.openqa.selenium.remote._
+
+import java.util
+
+import scala.collection.JavaConversions._
+
+object Firefox extends SeleniumBrowser {
+  def name: String = "Firefox"
+
+  def newDriver: BrowserDriver = new FirefoxDriver
+
+  override def initFiles(): Seq[VirtualJSFile] =
+    hijackConsole() +: super.initFiles()
+
+  /** The default selenium console output is broken, this is the workaround. */
+  private def hijackConsole(): VirtualJSFile = {
+    new MemVirtualJSFile("hijackConsole.js").withContent(
+      """
+        |(function () {
+        |  var console_log_hijack = [];
+        |  var oldLog = console.log;
+        |  console.log = function (msg) {
+        |    console_log_hijack.push(msg);
+        |    oldLog.apply(console, arguments);
+        |  };
+        |  this.scalajsPopHijackedConsoleLog = function () {
+        |    var log = console_log_hijack;
+        |    if (log.length != 0)
+        |      console_log_hijack = [];
+        |    return log;
+        |  };
+        |})();
+      """.stripMargin
+    )
+  }
+
+  private class FirefoxDriver extends BrowserDriver {
+    protected def newDriver(): RemoteWebDriver =
+      new org.openqa.selenium.firefox.FirefoxDriver()
+
+    def newConsoleLogsIterator(): Iterator[String] = {
+      getWebDriver.executeScript("return scalajsPopHijackedConsoleLog();") match {
+        case log: util.ArrayList[_] =>
+          log.iterator().map(_.toString)
+
+        case _ =>
+          throw new Exception("Ill formatted message.")
```

Comment:
It might be better to change this to
```scala
case x => throw new Exception(s"Ill-formatted message: [$x]")
```
so that if anything goes wrong we have more info to help fix it.